lib/nbp/push_connector.rb

```diff
@@ -0,0 +1,141 @@
+# frozen_string_literal: true
+
+require 'concurrent'
+
+module Nbp
+  class PushConnector
+    include Singleton
+
+    class Error < StandardError; end
+    class SettingsError < Error; end
+    class ServerError < Error; end
+
+    def initialize
+      super
+      @token = Concurrent::ThreadLocalVar.new
+      @token_expiration = Concurrent::ThreadLocalVar.new
+      @connector_settings = Concurrent::ThreadLocalVar.new
```

Comment:
Are we expecting any write changes to the connector settings? Otherwise, I would recommend to initialize them once here and don't worry about a `Concurrent::ThreadLocalVar`.

Reply:
No, there should be no changes. I removed the thread local var here.